### DISCUSSION TITLE
Virtual Camera Setup with CameraCapture

### DISCUSSION THREAD

**Discussion by banningjoe** [bonsai-rx/bonsai#2029] [General]
Hi all,

I've been trying to port in video data in a few different ways into Bonsai. One method I'm interested in using is a virtual camera. I've been able to setup OBS's virtual camera and have it be recognized by other software (Zoom). However, Bonsai doesn't seem to recognize the virtual camera. Has anyone been able to make this approach work? 

Thanks!

Joe

**Comment by aspaNeuro** [comment]
Yes, and there is a caveat.  
1. CameraCapture from Bonsai Vision doen't detect the OBS camera.
2. Need to install Bonsai - Video package in order for the camera to be recognized. This uses windows native DirectShow to get the capture device handle.
3. Even though, when you start the capture you might not to be able to see any output from the camera which you should (this may be related to the specifics on how directshow is andling the virtual camera device).
4. For the camera to start streaming into bonsai you need to connect something more to it (in my case a grayscale node solved the problem).
5. After that you can get the original image from the device and show it.

**Reply by banningjoe** [reply]
Great! This worked!

Just as a note:
The node VideoCaptureDevice must be used here rather than the CameraCapture node.

Thanks!

**Reply by JO-PANDA** [reply]
Hello I tried doing the same but as you can the video is not coming up. How do I ensure that it is recognized by the windows system? 
![image](https://github.com/user-attachments/assets/5439c734-1453-472b-a70d-e1f4d6a2157f)
